src/pip/_internal/cli/req_command.py

```diff
@@ -61,8 +63,10 @@ def configure_tempdir_registry(registry: TempDirectoryTypeRegistry) -> None:
             registry.set_delete(t, False)
 
     def wrapper(
-        self: RequirementCommand, options: Values, args: list[Any]
-    ) -> int | None:
+        self: _CommandT,
+        options: Values,
+        args: list[str],
+    ) -> int:
```

Comment:
Note that there is no use of the `| None`, and in all cases the args is `list[str]`:

 * https://github.com/pypa/pip/blob/758a1724c42e7f4369106dac09a50dcb0270931d/src/pip/_internal/commands/lock.py#L92
 * https://github.com/pypa/pip/blob/758a1724c42e7f4369106dac09a50dcb0270931d/src/pip/_internal/commands/download.py#L77
 * https://github.com/pypa/pip/blob/758a1724c42e7f4369106dac09a50dcb0270931d/src/pip/_internal/commands/install.py#L281
 * https://github.com/pypa/pip/blob/758a1724c42e7f4369106dac09a50dcb0270931d/src/pip/_internal/commands/wheel.py#L103